src/output-json-alert.c

```diff
@@ -523,6 +524,13 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
             AlertJsonPacket(p, js);
         }
 
+        /* signature text */
+        if (json_output_ctx->flags & LOG_JSON_RULE) {
+            hjs = json_object_get(js, "alert");
+            BUG_ON(!json_is_object(hjs));
```

Comment:
this will need a proper check

Reply:
What's wrong about the check? When the "alert" key is not there it's a bug in the code before, isn't it?

Reply:
Ah I see now, the "alert" key might not be there in out of memory conditions. I will create an updated PR.

Reply:
yeah we generally try to allow for alloc issues. Thanks!

Reply:
Updated PR: https://github.com/inliniac/suricata/pull/2897